DeratControl.Domain/Entities/Facility.cs

```diff
@@ -17,5 +18,16 @@ private Facility()
             this.Perimeters = new List<Perimeter>();
             this.Reviews = new List<Review>();
         }
+        public void AddPerimeter(PerimeterType perimeterType, User user)
+        {
+            foreach (var perimeter in this.Perimeters)
+            {
+                if (perimeterType == perimeter.PerimeterType)
+                {
+                    throw new PerimeterAlreadyExistsException();
+                }
+            }
+            this.Perimeters.Add(new Perimeter(this.Id, this, perimeterType, user.Id));
```

Comment:
declare a variable for Perimeter like, 
var newPerimeter = new Perimeter(this.Id, this, perimeterType, user.Id);
and push into a collection
this.Perimeters.Add(newPerimeter);

Change the return type of this method from void to Perimeter.
return newPerimeter;